migration-engine/core/src/migration/datamodel_calculator.rs

```diff
@@ -155,3 +157,51 @@ fn apply_create_field(data_model: &mut Datamodel, step: &CreateField) {
 
     model.add_field(field);
 }
+
+fn apply_create_index(data_model: &mut Datamodel, step: &CreateIndex) {
+    let model = data_model.find_model_mut(&step.model).expect(&format!(
+        "The model {} does not exist in this Datamodel. It is not possible to create an index in it.",
+        step.model
+    ));
+
+    match model
+        .indexes
+        .iter()
+        .find(|index| index.name == step.name && index.is_unique == step.is_unique && index.fields == step.fields)
+    {
+        Some(_) => panic!(
+            "The index {:?} on fields ({:?}) of model {} already exists in this Datamodel. It is not possible to create it once more.",
+            step.name, step.fields, model.name,
+        ),
+        None => {
+            let index = IndexDefinition {
+                name: step.name.clone(),
+                fields: step.fields.clone(),
+                is_unique: step.is_unique,
+            };
+
+            model.add_index(index)
+        }
+    }
+}
+
+fn apply_delete_index(data_model: &mut Datamodel, step: &DeleteIndex) {
+    let model = data_model.find_model_mut(&step.model).expect(&format!(
+        "The model {} does not exist in this Datamodel. It is not possible to drop an index in it.",
+        step.model
+    ));
+
+    match model
+        .indexes
+        .iter()
+        .find(|index| index.name == step.name && index.is_unique == step.is_unique && index.fields == step.fields) {
```

Comment:
This predicate looks like the one from above. Would it be possible to centralise this logic without making it look too complicated? This is optional. You decide :)

Reply:
I think we might have missed an important edge case. Imagine a user goes from the first schema to the second schema.
**first.schema**
```graphql
model User {
  id String @id @default(cuid())
  firstName String
  lastName String
  @@unique([firstName,lastName])
}
```

**second.schema**
```graphql
model User {
  id String @id @default(cuid())
  firstName String
  lastName String
  @@unique([firstName,lastName], name: "MyIndexName")
}
```

So the user is adding a custom name. Are we able to update index names at all in the db? My gut feeling is that this is not possible. Our current logic would drop the index and recreate it. Do you think this is what we should do? If yes we should add this as a destructive change check since this can be every expensive.

Reply:
Apparently it's possible in postgres and mysql:

-  `ALTER INDEX {} RENAME TO {}` in postgres: https://www.postgresql.org/docs/11/sql-alterindex.html
- `RENAME INDEX old_index_name TO new_index_name` in mysql (starting with 5.7) https://stackoverflow.com/questions/1463363/how-do-i-rename-an-index-in-mysql

I haven't found anything for sqlite